service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateAzureVmStep.java

```diff
@@ -0,0 +1,250 @@
+      AzureCloudContext azureCloudContext,
+      CrlService crlService,
+      ControlledAzureVmResource resource,
+      ResourceDao resourceDao) {
+    this.azureConfig = azureConfig;
+    this.azureCloudContext = azureCloudContext;
+    this.crlService = crlService;
+    this.resource = resource;
+    this.resourceDao = resourceDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
+
+    final ControlledAzureIpResource ipResource =
+        resourceDao
+            .getResource(resource.getWorkspaceId(), resource.getIpId())
+            .castToControlledResource()
+            .castToAzureIpResource();
+
+    final ControlledAzureDiskResource diskResource =
+        resourceDao
+            .getResource(resource.getWorkspaceId(), resource.getDiskId())
+            .castToControlledResource()
+            .castToAzureDiskResource();
+
+    // TODO: remove once networks are a controlled resource
+    final String name = getAzureName("network");
+    final String subnetName = getAzureName("subnet");
+    final String addressCidr = "192.168.0.0/16";
+    final String subnetAddressCidr = "192.168.1.0/24";
+
+    try {
+      Network network = createNetwork(name, subnetName, addressCidr, subnetAddressCidr);
+
+      Disk existingAzureDisk =
+          computeManager
+              .disks()
+              .getByResourceGroup(
+                  azureCloudContext.getAzureResourceGroupId(), diskResource.getDiskName());
+
+      PublicIpAddress existingAzureIp =
+          computeManager
+              .networkManager()
+              .publicIpAddresses()
+              .getByResourceGroup(
+                  azureCloudContext.getAzureResourceGroupId(), ipResource.getIpName());
+
+      computeManager
+          .virtualMachines()
+          .define(resource.getVmName())
+          .withRegion(resource.getRegion())
+          .withExistingResourceGroup(azureCloudContext.getAzureResourceGroupId())
+          // TODO: network, use a real network
+          .withExistingPrimaryNetwork(network)
+          .withSubnet(subnetName)
+          .withPrimaryPrivateIPAddressDynamic()
+          .withExistingPrimaryPublicIPAddress(existingAzureIp)
+          // See here for difference between 'specialized' and 'general' LinuxCustomImage, the
+          // managed disk storage option being the key factor
+          // https://docs.microsoft.com/en-us/azure/virtual-machines/linux/imaging#generalized-and-specialized
+          .withSpecializedLinuxCustomImage(azureConfig.getCustomDockerImageId())
+          .withExistingDataDisk(existingAzureDisk)
+          .withTag("workspaceId", resource.getWorkspaceId().toString())
+          .withTag("resourceId", resource.getResourceId().toString())
+          .withSize(VirtualMachineSizeTypes.STANDARD_D2S_V3)
+          .create(
```

Comment:
Just curious, is there a setting for startup scripts? (Don't think we need it for the demo, but we will eventually)

Reply:
Not that I can see from a quick documentation scan

Reply:
think I saw there's a method but we can leave it out for now since we're not planning to do proxying in Q4